samcli/local/docker/lambda_build_container.py

```diff
@@ -0,0 +1,164 @@
+                "bind": container_dirs["manifest_dir"],
+                "mode": "ro"
+            }
+        }
+
+        env_vars = None
+        if log_level:
+            env_vars = {
+                "LAMBDA_BUILDERS_LOG_LEVEL": log_level
+            }
+
+        super(LambdaBuildContainer, self).__init__(
+            image,
+            cmd,
+            container_dirs["source_dir"],
+            source_dir,
+            additional_volumes=additional_volumes,
+            entrypoint=entry,
+            env_vars=env_vars)
+
+    @property
+    def executable_name(self):
+        return LambdaBuildContainer._BUILDERS_EXECUTABLE
+
+    @staticmethod
+    def _make_request(protocol_version,
+                      language,
+                      dependency_manager,
+                      application_framework,
+                      container_dirs,
+                      manifest_file_name,
+                      runtime,
+                      optimizations,
+                      options):
+
+        return json.dumps({
+            "jsonschema": "2.0",
+            "id": 1,
+            "method": "LambdaBuilder.build",
+            "params": {
+                "__protocol_version": protocol_version,
+                "capability": {
+                    "language": language,
+                    "dependency_manager": dependency_manager,
+                    "application_framework": application_framework
+                },
+                "source_dir": container_dirs["source_dir"],
+                "artifacts_dir": container_dirs["artifacts_dir"],
+                "scratch_dir": container_dirs["scratch_dir"],
+
+                # Path is always inside a Linux container. So '/' is valid
+                "manifest_path": "{}/{}".format(container_dirs["manifest_dir"], manifest_file_name),
+
+                "runtime": runtime,
+                "optimizations": optimizations,
+                "options": options,
+            }
+        })
+
+    @staticmethod
+    def _get_entrypoint(request_json):
+        return [LambdaBuildContainer._BUILDERS_EXECUTABLE, request_json]
+
+    @staticmethod
+    def _get_container_dirs(source_dir, manifest_dir):
+        """
+        Provides paths to directories within the container that is required by the builder
+
+        Parameters
+        ----------
+        source_dir : str
+            Path to the function source code
+
+        manifest_dir : str
+            Path to the directory containing manifest
+
+        Returns
+        -------
+        dict
+            Contains paths to source, artifacts, scratch & manifest directories
+        """
+        base = "/tmp/samcli"
+        result = {
+            "source_dir": "{}/source".format(base),
+            "artifacts_dir": "{}/artifacts".format(base),
+            "scratch_dir": "{}/scratch".format(base),
+            "manifest_dir": "{}/manifest".format(base)
+        }
+
+        if pathlib.PurePath(source_dir) == pathlib.PurePath(manifest_dir):
```

Comment:
Nice check!

Reply:
I spent 3 hours debugging an issue that stemmed from not having this check ;)